### PR TITLE
refactor(connector): use utility function to raise payment method not implemented errors

### PR DESCRIPTION
**Pull request by swangi-kumari**
## Type of Change

- [ ] Bugfix
- [ ] New feature
- [ ] Enhancement
- [x] Refactoring
- [ ] Dependency updates
- [ ] Documentation
- [ ] CI/CD

## Description

Issue: defaults case in match shouldn't be handled using "_"

Solution:  "_" will never let the developer know impact of his changes incase of new addition of enum variants. By having the all cases in match compiler itself will throw errors wherever new enums needs to be handled

(In this PR it is done for Stripe, Bluesnap, Worldpay, checkout.com, and zen connectors.)

### Additional Changes

- [ ] This PR modifies the API contract
- [ ] This PR modifies the database schema
- [ ] This PR modifies application configuration/environment variables

## How did you test it?
Screen Shot 2023-08-01 at 7 44 05 PM 2
Screen Shot 2023-08-02 at 12 28 35 PM 2
Screen Shot 2023-08-02 at 2 50 50 PM

## Checklist

- [x] I formatted the code `cargo +nightly fmt --all`
- [x] I addressed lints thrown by `cargo clippy`
- [x] I reviewed submitted code
- [ ] I added unit tests for my changes where possible
- [ ] I added a [CHANGELOG](/CHANGELOG.md) entry if applicable

### CODE REVIEW

**Review by SanchithHegde — APPROVED** [review]
Looks good to me!

By the way, you can use the `clippy::wildcard_enum_match_arm` clippy lint to identify wildcard match arms. You can temporarily add it to [`.cargo/config.toml`](https://github.com/juspay/hyperswitch/blob/c464cc510ded595ea846e7da95f60919614e2bd3/.cargo/config.toml) to try it out, but expect it to throw a lot of warnings.

File: crates/router/src/connector/stripe/transformers.rs [hunk 1]
```diff
@@ -968,7 +1080,20 @@ impl TryFrom<&payments::BankRedirectData> for StripeBillingAddress {
...
+            | payments::BankRedirectData::Blik { .. }
+            | payments::BankRedirectData::Interac { .. }
+            | payments::BankRedirectData::OnlineBankingCzechRepublic { .. }
+            | payments::BankRedirectData::OnlineBankingFinland { .. }
+            | payments::BankRedirectData::OnlineBankingPoland { .. }
+            | payments::BankRedirectData::OnlineBankingSlovakia { .. }
+            | payments::BankRedirectData::Trustly { .. }
+            | payments::BankRedirectData::OnlineBankingFpx { .. }
+            | payments::BankRedirectData::OnlineBankingThailand { .. } => {
+                Err(errors::ConnectorError::NotImplemented(
```
Comment (ArjunKarthik) [inline crates/router/src/connector/stripe/transformers.rs:1093]: Error is not thrown in this scenario, Please return `Ok(Self::default()),`

File: crates/router/src/connector/utils.rs [hunk 2]
```diff
@@ -74,6 +74,10 @@ pub trait RouterData {
     fn get_quote_id(&self) -> Result<String, Error>;
 }
 
+pub fn payment_method_error_message(connector: &str) -> String {
```
Comment (ArjunKarthik) [inline crates/router/src/connector/utils.rs:77]: rename this method to `get_unimplemented_payment_method_error_message()` you can take this in next PR

File: crates/router/src/connector/stripe/transformers.rs [hunk 3]
```diff
@@ -15,7 +15,7 @@ use uuid::Uuid;
 
 use crate::{
     collect_missing_value_keys,
-    connector::utils::{ApplePay, PaymentsPreProcessingData, RouterData},
+    connector::utils::{self as util_connector, ApplePay, PaymentsPreProcessingData, RouterData},
```
Comment (ArjunKarthik) [inline crates/router/src/connector/stripe/transformers.rs:18]: rename this as connector_utils

**Review by ArjunKarthik — CHANGES_REQUESTED** [review]

**Review by ArjunKarthik — DISMISSED** [review]

**Review by SanchithHegde — DISMISSED** [review]

**Review by ArjunKarthik — APPROVED** [review]

**Pull request merged**